lib/ProductOpener/Products.pm

```diff
@@ -1088,14 +1089,34 @@ sub store_product($$$) {
 		delete $product_ref->{owners_tags};
 	}
 
-	push @{$changes_ref}, {
+	my $change_ref = {
 		userid => $user_id,
 		ip => remote_addr(),
 		t => $product_ref->{last_modified_t},
 		comment => $comment,
 		rev => $rev,
 	};
 
+	# Allow apps to send the user agent as a form parameter instead of a HTTP header, as some web based apps can't change the User-Agent header sent by the browser
+	my $user_agent = remove_tags_and_quote(decode utf8=>param("User-Agent"))
+		|| remove_tags_and_quote(decode utf8=>param("user-agent"))
+		|| remove_tags_and_quote(decode utf8=>param("user_agent"))
+		|| user_agent();
+
+	if ((defined $user_agent) and ($user_agent ne "")) {
+		$change_ref->{user_agent} = $user_agent;
+	}
+
+	# Allow apps to send app_name, app_version and app_uuid parameters
+	foreach my $field (qw(app_name app_version app_uuid)) {
+		my $value = remove_tags_and_quote(decode utf8=>param($field));
+		if ((defined $value) and ($value ne "")) {
+			$change_ref->{$field} = $value;
+		}
+	}
```

Comment:
So we have three string parameters:
- `app_name`
- `app_version`
- `app_uuid`